changelog/pending/20240228--pkg--schema-newpluginloader-now-respects-pulumi_debug_providers-which-enables-pulumi-yaml-programs-to-work-correctly-with-this-feature.yaml

```
@@ -0,0 +1,4 @@
+changes:
+- type: feat
+  scope: pkg
+  description: schema.NewPluginLoader now respects PULUMI_DEBUG_PROVIDERS, which enables Pulumi YAML programs to work correctly with this feature
```

```suggestion
  description: Make schema.NewPluginLoader respect PULUMI_DEBUG_PROVIDERS, which enables Pulumi YAML programs to work correctly with this feature
```

(We're trying to make changelog entries consistent, PR for documenting that has just been approved https://github.com/pulumi/pulumi/pull/15560)